Users/kegangum/fix errors

kegangum opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [x] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

kegangum committed:
added swagger for missing endpoints of the Deployment Resource Provider

fixed CI failures

kegangum committed:
added swagger for 'actionplan/operation/attempts' , 'operationresults' endpoints and corrected Product Package Secrets endpoint

kegangum committed:
updated the pfx file name type

kegangum committed:
fix LintDiff erros

kegangum committed:
fix LintDiff erros

added locations endpoint and fixed other changes

added actionplans to custom-words.txt and uodated ProductSecret.json

updated locations endpoint response

kegangum committed:
added locations endpoint and fixed other changes

kegangum committed:
added actionplans to custom-words.txt and uodated ProductSecret.json

kegangum committed:
updated locations endpoint response

AutorestCI commented:
You don't have permission to trigger SDK Automation.
Please add yourself to ***Azure*** group from opensource portal if you are MSFT employee,
or please ask reviewer to add comment *** /openapibot sdkautomation ***.
Please ask tih@microsoft.com (or NullMDR in github) for additional help.

kegangum committed:
updated the location parameter to global

kegangum committed:
updated the location parameter to global

added 404 and default responses

added locations endpoint - list operation

removed 404 response for the 'ProductDeployment-remove operation'

Adding  'productDeployments/executeRunner/action''

removed 404 responses

added default error responses and deleted 404 responses for file containers

removed older version in the read.md 'input-file' section

changed details info in the "ExtendedErrorInfo"

changed ExtendedErrorInfo object

updating error definition

changing error definition in 2018-07-01 version

updated location responses

changed the entity name

Revert "changed the entity name"

This reverts commit 2c7e079dadbf5298640af902cec7f42f8ff442d9.

Revert "updated location responses"

This reverts commit 78bca1da75dad66409bfa6a950ca58ba02580b88.

updated locations Get response

change for New-AzFileContainer command

added Parameter for the New-AzsProductPackage

changed productId to packageId

updated operationId for Bootstrap

changed operationId for deploy action

changed paramters to object

changed the operation Ids

changed operationIds for the right command names

execute runner changes

fixed path parameter for secrets and remove command name

changed the secret name

changed secret parameters

added changes to resolve conflicts

changed version number

changed the version number

changed the tag name

to fix conflicts

azuresdkci commented:
Can one of the admins verify this patch?

kegangum commented:
The result from LintDiff is wrong as RP name is already in PascalCase . Breaking Change can be ignoreed.

bganapa commented:
@njuCZ Could you please review this PR?

bganapa commented on specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2018-07-01/FileContainer.json:
```diff
@@ -0,0 +1,363 @@
+              "$ref": "#/definitions/FileContainer"
+            }
+          },
+          "404": {
```
All the 404 should be repalced with Default error response

bganapa replied:
like this https://github.com/Azure/azure-rest-api-specs/blob/master/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2019-12-01/apimapis.json#L100

kegangum replied:
added Default error responses . Please review it .

bganapa commented on specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2018-07-01/examples/FileContainer/Delete.json:
```diff
@@ -0,0 +1,12 @@
+  },
+  "responses": {
+    "200": {},
+    "404": {}
```
pls remove 404s

kegangum replied:
removed

bganapa commented on specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ProductSecret.json:
```diff
@@ -143,7 +143,7 @@
     },
-    "/subscriptions/{subscriptionId}/providers/Microsoft.Deployment.Admin/locations/global/productSecrets/{productId}/secrets/{secretName}/validate": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Deployment.Admin/locations/global/productPackages/{productId}/secrets/{secretName}/validate": {
       "post": {
```
is this correct? secrets comes under ProductPackages?

kegangum replied:
yes , secrets are coming from productPackages

KrisBash reviewed (approved):
No API changes for ARM to review

kegangum committed:
added 404 and default responses

kegangum committed:
added locations endpoint - list operation

bganapa commented on specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/examples/ProductDeployment/Remove.json:
```diff
@@ -7,6 +7,7 @@
     "200": {},
-    "202": {}
+    "202": {},
+    "404": {}
```
404 responses should not be there , pls remove

bganapa commented on specification/azsadmin/resource-manager/deployment/readme.md:
```diff
@@ -72,10 +80,15 @@ input-file:
   - $(this-folder)/Microsoft.Deployment.Admin/preview/2019-01-01/ProductPackage.json
   - $(this-folder)/Microsoft.Deployment.Admin/preview/2019-01-01/ProductSecret.json
+  - $(this-folder)/Microsoft.Deployment.Admin/preview/2019-01-01/OperationResults.json
+  - $(this-folder)/Microsoft.Deployment.Admin/preview/2018-07-01/FileContainer.json
```
Product Package and File Container are supported only with 2018-07-01 api version?

kegangum replied:
Product Package and File Container are supported both in '2018-07-01' and '2019-01-01' versions

bganapa replied:
Lets include the 2019-01-01 version here then. Do we have any changes between those api versions for ProductPackage and FielContainer?

kegangum replied:
I have already included 2019-01-01 and 2018-07-01 versions of FileContainer and ProductPackage in the 'read.md' file. Please review it . We don't have any changes for the both the versions . Few ARM services are still calling older versions of FileContainer and ProductPackages .

bganapa replied:
Here the version should be the latest 2019-0-01 instead of 2018-07-01. you can keep the 2018 version in a older tag. in the input-file: lets use 2019-01-01. This tag is used for azure powershell autogeneration

kegangum replied:
updated

kegangum committed:
removed 404 response for the 'ProductDeployment-remove operation'

kegangum committed:
Adding  'productDeployments/executeRunner/action''

bganapa commented on specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2018-07-01/examples/ProductPackage/List.json:
```diff
@@ -0,0 +1,27 @@
+        ]
+      }
+    },
+    "404": {}
```
pl remove 404s everywhere

bganapa commented on specification/azsadmin/resource-manager/deployment/Microsoft.Deployment.Admin/preview/2019-01-01/ActionPlanOperation.json:
```diff
@@ -45,6 +45,15 @@
               "$ref": "#/definitions/ActionPlanOperationsList"
             }
+          },
+          "404": {
```
404 error response needs to be removed everywhere, having default error response is enough

kegangum committed:
removed 404 responses

kegangum committed:
added default error responses and deleted 404 responses for file containers

kegangum committed:
removed older version in the read.md 'input-file' section

kegangum committed:
changed details info in the "ExtendedErrorInfo"

kegangum committed:
changed ExtendedErrorInfo object

kegangum committed:
updating error definition

kegangum committed:
changing error definition in 2018-07-01 version

kegangum commented:
This PR is good to merge . Please ignore Breaking change error and PathResourceProviderNamePascalCase error.

kegangum committed:
Merge remote-tracking branch 'upstream/master' into users/kegangum/fixErrors

bganapa commented:
@njuCZ Could you please merge this PR? The breaking changes are expected as we are in preview and we are really fixing the bugs in swagger files

kegangum committed:
updated location responses

kegangum committed:
changed the entity name

kegangum committed:
Revert "changed the entity name"

This reverts commit 2c7e079dadbf5298640af902cec7f42f8ff442d9.

kegangum committed:
Revert "updated location responses"

This reverts commit 78bca1da75dad66409bfa6a950ca58ba02580b88.

kegangum committed:
updated locations Get response

njuCZ commented:
@bganapa 
Per [Retirement of Preview guidance](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews), products in Preview for over one year need to move to GA or retire; require CVP exemption after 1 year in preview.

bganapa commented:
@njuCZ Could you please ping me(bganapa) on Teams? I could not find your microsoft id. 

We have a plan to move our specs out of preview . We need to get this PR merged now without waiting for moving them out of preview.

kegangum committed:
change for New-AzFileContainer command

kegangum committed:
added Parameter for the New-AzsProductPackage

kegangum committed:
changed productId to packageId

kegangum committed:
updated operationId for Bootstrap

kegangum committed:
changed operationId for deploy action

bganapa commented:
@njuCZ , We will be moving all our specs to stable in Fe timeframe. Could you please go ahead and merge this?

kegangum committed:
changed paramters to object

kegangum committed:
changed the operation Ids

kegangum committed:
changed operationIds for the right command names

njuCZ commented:
/azp run

kegangum committed:
execute runner changes

kegangum committed:
fixed path parameter for secrets and remove command name

kegangum committed:
changed the secret name

kegangum committed:
changed secret parameters

kegangum committed:
Merge pull request #1 from kegangum/users/kegangum/autorest

Users/kegangum/autorest

kegangum committed:
added changes to resolve conflicts

kegangum committed:
changed version number

kegangum committed:
changed the version number

kegangum committed:
changed the tag name

kegangum committed:
to fix conflicts

kegangum committed:
Merge branch 'users/kegangum/fixErrors' of https://github.com/kegangum/azure-rest-api-specs into users/kegangum/fixErrors

akning-ms commented:
Hi @Kegangum, The Linter  failure for PathResourceProviderNamePascalCase can be ignored as it is due to Microsoft.Deployment.Admin has 2 dot which is not like others we are not support it in tool. 
But it created new linter error: R4011 - DeleteOperationResponses  can you help fix it?

kegangum committed:
added 204 responses for delete operation
